### PR TITLE
Kwabena/ov2640 fixes

### PR DESCRIPTION
**Pull request by kwagyeman**
* Images are right side up now.
* Picture quality is acceptable (not as good as the OV7725)
* All auto functions work now (the OV2640 ignores exposure control
however)
* Added XGA frame size.
* JPEG mode is enabled but needs work still (not sure if the H7 hardware
can capture the packet stream fast enough not to drop bytes).
* Fix bayer/jpeg image print statement.

### CODE REVIEW

File: src/omv/sensor.c [hunk 1]
```diff
@@ -996,7 +1000,13 @@ int sensor_snapshot(sensor_t *sensor, image_t *image, streaming_cb_t streaming_c
                 break;
             case PIXFORMAT_JPEG:
                 // Read the number of data items transferred
-                MAIN_FB()->bpp = (MAX_XFER_SIZE - __HAL_DMA_GET_COUNTER(&DMAHandle))*4;
+                MAIN_FB()->bpp = ((MAX_XFER_SIZE/4) - __HAL_DMA_GET_COUNTER(&DMAHandle))*4;
+                // Remove trailing zeros...
```
Comment (iabdalkader) [inline src/omv/sensor.c]: Why is this needed ?

Comment (kwagyeman) [reply src/omv/sensor.c]: Removes the trailing zeros in a jpeg image if there are any,
